lib/multihandle.c

```diff
@@ -0,0 +1,375 @@
+/*
+ * Copyright (C) 1995,1996,1997 Lars Fenneberg
+ *
+ * Copyright 1992 Livingston Enterprises, Inc.
+ *
+ * Copyright 1992,1993, 1994,1995 The Regents of the University of Michigan
+ * and Merit Network, Inc. All Rights Reserved
+ *
+ * Copyright (C) 2022 Cadami GmbH, info@cadami.net
+ *
+ * See the file COPYRIGHT for the respective terms and conditions.
+ * If the file is missing contact me at lf@elemental.net
+ * and I'll send you a copy.
+ */
+
+#include <includes.h>
+#include <radcli/radcli.h>
+#include <pathnames.h>
+#include <poll.h>
+
+#include "sendserver_util.h"
+#include "util.h"
+#include "rc-md5.h"
+#include "rc-hmac.h"
+
+#if defined(__linux__)
+#include <linux/in6.h>
+#endif
+
+struct rc_async_handle_list_member {
+	struct rc_async_handle_list_member *previous, *next;
+	struct rc_async_handle *hdl;
+};
+
+
+struct rc_multihandle {
```

Comment:
It is not clear to me what is the intention of the multihandle. Some documentation is needed here to explain why this exists and what it does.